src/types/opt_arg.rs

```diff
@@ -0,0 +1,55 @@
+use std::fmt;
+
+use crate::NativeRepr;
+
+/// A convenience type to explicitly mark
+/// a function argument as `opt T`.
+///
+/// > When left unspecified on Redscript side,
+/// it translates to its `Default` representation.
+#[derive(Default)]
+pub enum OptArg<T: NativeRepr + Default + PartialEq> {
```

Comment:
```suggestion
pub enum OptArg<T: NativeRepr + Default + PartialEq> {
```
putting bounds on data structures is discouraged: https://rust-lang.github.io/api-guidelines/future-proofing.html#data-structures-do-not-duplicate-derived-trait-bounds-c-struct-bounds

you can put them everywhere else, but I'd remove them from here
also I think you should be able to derive Debug, Clone and Copy

Reply:
Ok, let me refactor in a bit. I don't think we should add `Copy` unless `T` is itself `Copy` (some types aren't iirc).

Reply:
yes, but the derived `Copy` will be constrained by `Copy` on `T` automatically

Reply:
At the same time for `NativeRepr + Default + PartialEq` these are the minimal guarantees for `T` to be accepted as an `OptArg<T>` / `opt T`.

Reply:
that's true, but the user can't misuse it anyway as long as the bounds exist on `FromRepr`,
putting the bounds on the type just forces us to use the bounds everywhere else

Reply:
so is there any misding change on my side @jac3km4 or I can close this review?